### PR TITLE
[ELF] Rename IsRela to HasAddend

### PR DESCRIPTION
**Pull request by MaskRay**
`IsRela` is used by lld to differentiate REL and RELA static
relocations. The proposed CREL patch will reuse `IsRela` for CREL
(#91280). Rename `IsRela` to be more appropriate.

### CODE REVIEW

File: lld/ELF/InputSection.cpp [hunk 1]
```diff
@@ -1010,7 +1010,7 @@ void InputSection::relocateNonAlloc(uint8_t *buf, ArrayRef<RelTy> rels) {
     // For a relocatable link, content relocated by RELA remains unchanged and
     // we can stop here, while content relocated by REL referencing STT_SECTION
     // needs updating implicit addends.
-    if (config->relocatable && (RelTy::IsRela || sym.type != STT_SECTION))
+    if (config->relocatable && (RelTy::HasAddend || sym.type != STT_SECTION))
```
Comment (smithp35) [inline lld/ELF/InputSection.cpp:1013]: I think the comment above might be worth updating assuming CREL will behave in the same way. Perhaps something like:
```
For a relocatable link, content relocated by relocation types with an explicit addend, such as RELA, remain unchanged and we can stop here. While content relocated by relocation types with an implicit addend, such as REL, needs the implicit addend updated.
```

File: lld/ELF/Relocations.cpp [hunk 2]
```diff
@@ -492,7 +492,7 @@ int64_t RelocationScanner::computeMipsAddend(const RelTy &rel, RelExpr expr,
 
   // The ABI says that the paired relocation is used only for REL.
   // See p. 4-17 at ftp://www.linux-mips.org/pub/linux/mips/doc/ABI/mipsabi.pdf
-  if (RelTy::IsRela)
+  if (RelTy::HasAddend)
```
Comment (smithp35) [inline lld/ELF/Relocations.cpp:495]: Assuming MIPS will support CREL, but the ABI won't be updated to explicitly mention it, could be worth a small addition at the end.
```
The ABI says the paired relocation  is used only for REL.
  // See p. 4-17 at ftp://www.linux-mips.org/pub/linux/mips/doc/ABI/mipsabi.pdf this generalises to relocation types with implicit addends.
```

**Review by smithp35 — COMMENTED** [review]
No objections from me, a couple of places where some adjacent comments might be worth updating. Not got a particularly strong opinion on that though.

**Review by jh7370 — APPROVED** [review]
LGTM, pending @smithp35's comments. (I note the buildkite failure, but assume it to be unrelated)

File: lld/ELF/Relocations.cpp (hunk 2 above)
Comment (MaskRay) [reply lld/ELF/Relocations.cpp:495]: Thx. Added. (I think MIPS will die before anyone implements CREL for it...)

**Pull request merged**